apps/studio/components/interfaces/SchemaVisualizer/index.tsx

```diff
@@ -0,0 +1,166 @@
+import { PGlite } from '@electric-sql/pglite'
+import { PostgresMetaBase, wrapError, wrapResult } from '@gregnr/postgres-meta/base'
+import { useEffect, useRef, useState } from 'react'
+import { Edge, Node, ReactFlowProvider } from 'reactflow'
+
+import { SchemaFlow } from 'components/interfaces/ProjectCreation/SchemaFlow'
+import { getGraphDataFromTables } from '../Database/Schemas/Schemas.utils'
+import { TableNodeData } from '../Database/Schemas/SchemaTableNode'
+
+export const TABLE_NODE_WIDTH = 640
+export const TABLE_NODE_ROW_HEIGHT = 80
+
+interface SchemaVisualizerProps {
+  sqlStatements: string[]
+  className?: string
+}
+
+export const SchemaVisualizer = ({ sqlStatements, className }: SchemaVisualizerProps) => {
+  const [nodes, setNodes] = useState<Node<TableNodeData>[]>([])
+  const [edges, setEdges] = useState<Edge[]>([])
+  const db = useRef<PGlite>()
+  const executedStatements = useRef<Set<string>>(new Set())
+
+  // Initialize database once
+  useEffect(() => {
+    db.current = new PGlite()
+
+    // Execute initial auth schema setup
+    db.current.exec(`
+      CREATE SCHEMA auth;
+      CREATE TABLE auth.users (
+          instance_id uuid,
+          id uuid NOT NULL,
+          aud character varying(255),
+          role character varying(255),
+          email character varying(255),
+          encrypted_password character varying(255),
+          confirmed_at timestamp with time zone,
+          invited_at timestamp with time zone,
+          confirmation_token character varying(255),
+          confirmation_sent_at timestamp with time zone,
+          recovery_token character varying(255),
+          recovery_sent_at timestamp with time zone,
+          email_change_token character varying(255),
+          email_change character varying(255),
+          email_change_sent_at timestamp with time zone,
+          last_sign_in_at timestamp with time zone,
+          raw_app_meta_data jsonb,
+          raw_user_meta_data jsonb,
+          is_super_admin boolean,
+          created_at timestamp with time zone,
+          updated_at timestamp with time zone
+      );
+      ALTER TABLE ONLY auth.users
+      ADD CONSTRAINT users_email_key UNIQUE (email);
+      ALTER TABLE ONLY auth.users
+      ADD CONSTRAINT users_pkey PRIMARY KEY (id);
```

Comment:
this seems duplicate with below at L75 - could we pull this out into a const?
okay for this to be done in another PR